### PR TITLE
RF-5179 Fix hashing / update according to latest ruby version

### PR DESCRIPTION
**Pull request by jkaipr**
We haven't created this library, but since there are clients out there using it, we decided to publish a fix so they can continue using it.

This PR ports changes from the original ruby version of Rainforest auth. This PR in the ruby version https://github.com/rainforestapp/auth/pull/2/files contains the changes I've ported over.

I've added unit tests that are equal to the ruby version to make sure the implementation behavior is exactly the same.

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      expect(urlSplit[7]).to.be.equal(runId);
+    });
+
+    it('has the correct action', () => {
+      expect(urlSplit[8]).to.be.equal('before_run');
+    });
+
+    it('has the correct digest', () => {
+      const digest = urlSplit[9];
+      console.log('digest', digest);
```
Comment (gesposito) [inline test/index.spec.js]: Do we get value out of this log?

File: test/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  });
+
+  describe('get_run_callback', () => {
+    let url, urlSplit;
+    const runId = '123456';
+
+    beforeEach(() => {
+      rfAuth = new auth(key);
+      url = rfAuth.get_run_callback(runId, 'before_run');
+      console.log(url);
```
Comment (gesposito) [inline test/index.spec.js]: I guess we could avoid this as well

File: test/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      const digest = urlSplit[9];
+      console.log('digest', digest);
+      const verified = rfAuth.verify(digest, 'before_run', { run_id: runId });
+      expect(verified).to.be.true;
+    });
+  });
+
+  describe('sign', () => {
+    it('returns the expected signature', () => {
+      expect(rfAuth.sign('test', { option: 1 })).to.be.equal('65f2253344287b3c5634a1ce6163fb694b2280b1');
```
Comment (gesposito) [inline test/index.spec.js]: I'm tempted to share these 4 digests as `const`s in order to make it clear that they glue sign/sign_old/verify steps together.

**Review by gesposito — APPROVED** [review]
Just left some minor nitpicks, LGTM

File: test/index.spec.js (hunk 1 above)
Comment (jkaipr) [reply test/index.spec.js]: nope, just a leftover

File: test/index.spec.js (hunk 3 above)
Comment (jkaipr) [reply test/index.spec.js]: good point, I'll update it

**Pull request merged**